nym-vpn-apple/NymMixnetTunnel/PacketTunnelProvider.swift

```diff
@@ -4,100 +4,277 @@ import NymLogger
 import MixnetLibrary
 import TunnelMixnet
 import Tunnels
+import Constants
+
+enum TunnelEvent {
+    case statusUpdate(TunStatus)
+    case runFailure(Error)
+}
 
 class PacketTunnelProvider: NEPacketTunnelProvider {
-    private lazy var mixnetTunnelProvider = MixnetTunnelProvider()
-    private lazy var mixnetAdapter: MixnetAdapter = {
-        MixnetAdapter(
-            with: self,
-            mixnetTunnelProvider: mixnetTunnelProvider
-        )
-    }()
     private lazy var logger = Logger(label: "MixnetTunnel")
 
+    private let eventStream: AsyncStream<TunnelEvent>
+    private let eventContinuation: AsyncStream<TunnelEvent>.Continuation
+
     override init() {
         LoggingSystem.bootstrap { label in
-            FileLogHandler(label: label)
+            let fileLogHandler = FileLogHandler(label: label)
+            
+            #if DEBUG
+                let osLogHandler = OSLogHandler(subsystem: Bundle.main.bundleIdentifier!, category: label);
+                return MultiplexLogHandler([osLogHandler, fileLogHandler])
+            #else
+                return fileLogHandler
+            #endif
         }
+
+        let (eventStream, eventContinuation) = AsyncStream<TunnelEvent>.makeStream();
+        self.eventStream = eventStream
+        self.eventContinuation = eventContinuation
     }
 
     override func startTunnel(options: [String: NSObject]? = nil) async throws {
-        logger.log(level: .info, "Start tunnel...")
-        guard
-            let tunnelProviderProtocol = self.protocolConfiguration as? NETunnelProviderProtocol,
-            let mixnetConfig = tunnelProviderProtocol.asMixnetConfig()
-        else {
-            logger.log(level: .info, "Start tunnel: invalid saved configuration")
+        logger.info("Start tunnel...")
+
+        setup()
+
+        guard let tunnelProviderProtocol = protocolConfiguration as? NETunnelProviderProtocol,
+              let mixnetConfig = tunnelProviderProtocol.asMixnetConfig() else {
+            logger.error("Failed to obtain tunnel configuration")
             throw PacketTunnelProviderError.invalidSavedConfiguration
         }
 
-        let callback: () -> Void = { [weak self] in
-            guard let config = self?.mixnetTunnelProvider.nymConfig
-            else {
-                return
-            }
+        var vpnConfig = try mixnetConfig.asVpnConfig(tunProvider: self, tunStatusListener: self)
+        // todo: figure out WHAT always defaults this setting to .randomLowLatency
```

Comment:
https://github.com/nymtech/nym-vpn-client/blob/94ba6cda9598710f42c8f17935de9dbb7642dadd/nym-vpn-apple/Services/Sources/Services/ConnectionManager/ConnectionStorage.swift#L29

Changing this to `.random`

Reply:
or here https://github.com/nymtech/nym-vpn-client/blob/97db4f4eb056d62fc5e725acda6f0423d23f366c/nym-vpn-apple/Services/Sources/Services/TunnelMixnet/Config/MixnetConfig.swift#L67

Reply:
Rokas have you already fixed that right?

Reply:
Should be working in any case after this was merged:
https://github.com/nymtech/nym-vpn-client/pull/885